code/game/objects/structures/stool_bed_chair_nest/bed.dm

```diff
@@ -643,15 +705,65 @@ GLOBAL_LIST_EMPTY(activated_medevac_stretchers)
 			playsound(loc,'sound/machines/buzz-two.ogg', 25, FALSE)
 			return
 
-		var/obj/item/roller/medevac/R = I
-		linked_bed = R
-		R.linked_beacon = src
-		to_chat(user, span_notice("<b>You link the medvac beacon to the medvac stretcher.</b>"))
-		playsound(loc,'sound/machines/ping.ogg', 25, FALSE)
+		add_stretcher(I, user)
 
 
 /obj/item/medevac_beacon/proc/check_power()
 	var/area/A = loc?.loc
 	if(!A || !isarea(A))
 		return FALSE
 	return(A.powered(1))
+
+/// Adds a medevac roller or stretcher to the medevac beacon. Returns TRUE if the beacon is in the linked_beds* list and false if it is not in there.
+/obj/item/medevac_beacon/proc/add_stretcher(obj/O, mob/user, silent = FALSE)
+	var/obj/item/roller/medevac/R = O
+	if(R && istype(R, /obj/item/roller/medevac))
```

Comment:
istype should account for there not being an `R`. Also chill with the single letter var names, make them descriptive.